extras/ioc_scan_by_host.sh

```diff
@@ -50,15 +50,16 @@ echo Instances are: "${instances[*]}"
 # Loop through all instance IDs
 for instance_id in "${instances[@]}"; do
   echo
-  echo Searching "$instance_id":
+  echo -n Searching "$instance_id":
 
-  # Use find-grep to search for IOC strings in log files, ignoring
-  # *.journal files.  We pipe the result into another grep process
-  # that uses the --invert-match grep flag to exclude matches (e.g.,
-  # from sudo.log) that contain our grep command (e.g. sudo.log).
+  # Use find-zgrep to search for IOC strings in log files (including
+  # gzipped log files), ignoring *.journal files.  We pipe the result
+  # into another grep process that uses the --invert-match grep flag
+  # to exclude matches (e.g., from sudo.log) that contain our grep
+  # command (e.g. sudo.log).
   aws ssm start-session --target="$instance_id" \
     --document=AWS-StartInteractiveCommand \
-    --parameters="command='for i in ${iocList[*]}; do sudo find /var/log -type f -not -name \*\.journal -exec grep --ignore-case --recursive \$i {} \; | grep --invert-match -- --ignore-case\ --recursive\ | echo \$(wc --lines) found for \$i; done'"
+    --parameters="command='hostname; for i in ${iocList[*]}; do sudo find /var/log -type f -not -name \*\.journal -exec zgrep --ignore-case \$i {} \; | grep --invert-match -- grep\ --ignore-case | echo \$(wc --lines) found for \$i; done'"
```

Comment:
@bra1ncramp When we discussed this in chat, we said that wanted to exclude matches from previous runs of this script (when our command was `grep` and not `zgrep`), but since that is just a short-term situation, I think it makes more sense to just change this for the long term:
```suggestion
    --parameters="command='hostname; for i in ${iocList[*]}; do sudo find /var/log -type f -not -name \*\.journal -exec zgrep --ignore-case \$i {} \; | grep --invert-match -- zgrep\ --ignore-case | echo \$(wc --lines) found for \$i; done'"
```

Otherwise, we will just confuse our future selves and future maintainers of this code.

Reply:
That's fair. I had the same exact thought, to be honest.